src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteHandlers/RouteHandlerAnalyzer.cs

```diff
@@ -77,8 +77,11 @@ public override void Initialize(AnalysisContext context)
                     var foundMethodReferenceBody = false;
                     if (!methodReference.Method.DeclaringSyntaxReferences.IsEmpty)
                     {
-                        var syntaxReference = methodReference.Method.DeclaringSyntaxReferences[0];
-                        var methodOperation = invocation.SemanticModel.GetOperation(syntaxReference.GetSyntax(context.CancellationToken));
+                        var syntaxReference = methodReference.Method.DeclaringSyntaxReferences.SingleOrDefault();
```

Comment:
nit: I think `Single` should be okay since we already checked `DeclaringSyntaxReferences` is not empty.